spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/messaging/SleuthKafkaStreamsConfigurationTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.instrument.messaging;
+
+import brave.Tracing;
+import org.apache.kafka.streams.KafkaClientSupplier;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.config.StreamsBuilderFactoryBean;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.BDDAssertions.then;
+
+/**
+ * @author Tim te Beek
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = SleuthKafkaStreamsConfigurationTest.Config.class,
+		webEnvironment = WebEnvironment.NONE)
+public class SleuthKafkaStreamsConfigurationTest {
+
+	@Autowired
+	TestTraceStreamsBuilderFactoryBean streamsBuilderFactoryBean;
+
+	@Test
+	public void clientSupplierInvokedOnStreamsBuilderFactoryBean() {
```

Comment:
I would add javadoc about why it is important that StreamsBuilderFactoryBean has the tracing supplier. Maybe mention the use case that now works. cc @jeqo

Reply:
Added documentation in src/main in d8338ba0cb0ef2dca37e11b2de6a8ef5b7925a79, hope that's clear enough to resolve this item.

Reply:
yep